### PR TITLE
change(ci): run coverage on stable

### LINKED ISSUE #3887: Move coverage builds to stable Rust
## Motivation

In PR #3886, we disabled coverage builds, because nightly Rust had bugs.

Stable Rust now supports coverage, and it will be more reliable than nightly.

## Tasks

Coverage was stabilised in Rust 1.60:
https://github.com/rust-lang/rust/blob/master/RELEASES.md#compiler

- Switch the coverage job to stable Rust
- Switch to using `-C instrument-coverage`

## Follow Up

Upgrade the coverage action:
- #4043

### PR DESCRIPTION
**Pull request by dconnolly**
## Motivation

We've been running our coverage collection on nightly Rust as that was the only access to the source-based coverage support to give the best / most reliable data. 

## Solution

Source-based coverage support has been stabilized and the cargo component we use to wrap up the compiler flags has updated accordingly. This PR switches our Coverage workflow to Rust stable, which provides more reliable stability guarantees in general for our CI.

Closes #3887 

## Review

Anyone can review, low priority.

The `coverage.patch.yml` workflow is a no-op so there is no Rust channel to update there.

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -36,9 +36,8 @@ env:
 
 jobs:
   coverage:
-    name: Coverage nightly
+    name: Coverage (+stable)
```
Comment (teor2345) [inline .github/workflows/coverage.yml]: Nitpick: can we match the other job names that mention the Rust version?

(Or we could just say "Coverage".)

```suggestion
    name: Coverage on stable
```

Comment (dconnolly) [reply .github/workflows/coverage.yml]: Old habits :D

Comment (gustavovalverde) [reply .github/workflows/coverage.yml]: Sorry for damaging those habits 😆

**Review by gustavovalverde — CHANGES_REQUESTED** [review]
Leaving this comment here so we don't forget about the patch file.

**Comment by dconnolly** [comment]
> Leaving this comment here so we don't forget about the patch file.

I looked at `coverage.patch.yml` and it's a no-op, no Rust toolchain to update

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -36,9 +36,8 @@ env:
 
 jobs:
   coverage:
-    name: Coverage nightly
+    name: Coverage on stable
     # The large timeout is to accommodate:
-    # - nightly builds (75 minutes, typically 30-50 minutes), and
     # - parameter downloads (an extra 90 minutes, but only when the cache expires)
     timeout-minutes: 165
```
Comment (oxarbitrage) [inline .github/workflows/coverage.yml]: Should we change this to 140 minutes ? 
```suggestion
    timeout-minutes: 140
```

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -36,9 +36,8 @@ env:
 
 jobs:
   coverage:
-    name: Coverage nightly
+    name: Coverage on stable
     # The large timeout is to accommodate:
-    # - nightly builds (75 minutes, typically 30-50 minutes), and
```
Comment (oxarbitrage) [inline .github/workflows/coverage.yml:41]: Instead of deleting we should have something as:

`# - stable builds (typically 30-50 minutes), and`

**Comment by teor2345** [comment]
> > Leaving this comment here so we don't forget about the patch file.
> 
> I looked at `coverage.patch.yml` and it's a no-op, no Rust toolchain to update

The job name in the patch file needs to be changed, so we can require the job in branch protection rules, but skip it when it is not needed:
https://github.com/ZcashFoundation/zebra/blob/e4ca6c358ec9807c5c97e610d51297a0dab7996d/.github/workflows/coverage.patch.yml#L16

File: .github/workflows/coverage.yml [hunk 4]
```diff
@@ -36,9 +36,8 @@ env:
 
 jobs:
   coverage:
-    name: Coverage nightly
+    name: Coverage on stable
```
Comment (teor2345) [inline .github/workflows/coverage.yml:39]: > > Leaving this comment here so we don't forget about the patch file.
> 
> I looked at `coverage.patch.yml` and it's a no-op, no Rust toolchain to update

The job name in the patch file needs to be changed, so we can require the job in branch protection rules, but skip it when it is not needed:
https://github.com/ZcashFoundation/zebra/blob/e4ca6c358ec9807c5c97e610d51297a0dab7996d/.github/workflows/coverage.patch.yml#L16

File: .github/workflows/coverage.yml (hunk 3 above)
Comment (gustavovalverde) [reply .github/workflows/coverage.yml:41]: Included in [fix(ci): timeout and docs](https://github.com/ZcashFoundation/zebra/pull/4465/commits/9f6ddb7bb9082a8084b1933a9075f15a7cb41d04)

File: .github/workflows/coverage.yml (hunk 2 above)
Comment (gustavovalverde) [reply .github/workflows/coverage.yml]: Considered in [fix(ci): timeout and docs](https://github.com/ZcashFoundation/zebra/pull/4465/commits/9f6ddb7bb9082a8084b1933a9075f15a7cb41d04)

File: .github/workflows/coverage.yml (hunk 4 above)
Comment (gustavovalverde) [reply .github/workflows/coverage.yml:39]: Fixed with [fix(ci): sync patch file](https://github.com/ZcashFoundation/zebra/pull/4465/commits/5697ee9eae1d83441cb181c4e06d105b65590e70)

**Review by gustavovalverde — APPROVED** [review]

**Comment by gustavovalverde** [comment]
@Mergifyio update

**Review by teor2345 — APPROVED** [review]
Thanks!

**Comment by teor2345** [comment]
@mergifyio requeue

**Comment by teor2345** [comment]
@mergifyio update

**Pull request merged**